src/tx/penalty/transaction.ts

```diff
@@ -128,6 +128,43 @@ function recordPenaltyTX(txId: string, tx: PenaltyTX): void {
   }
 }
 
+/**
+ * Compares the event timestamp of the penalty tx with the timestamp of the last saved penalty tx
+ */
+function isDuplicatePenaltyTx(
+  tx: PenaltyTX,
+  nodeAccount: NodeAccount2
+): { isDuplicate: boolean; eventTime: number } {
+  switch (tx.violationType) {
+    case ViolationType.LeftNetworkEarly:
+      return {
+        isDuplicate:
+          nodeAccount.nodeAccountStats.lastPenaltyTime ===
+          (tx.violationData as LeftNetworkEarlyViolationData).nodeDroppedTime,
+        eventTime: (tx.violationData as LeftNetworkEarlyViolationData).nodeDroppedTime,
+      }
+
+    case ViolationType.NodeRefuted:
+      return {
+        isDuplicate:
+          nodeAccount.nodeAccountStats.lastPenaltyTime ===
+          (tx.violationData as NodeRefutedViolationData).nodeRefutedTime,
+        eventTime: (tx.violationData as NodeRefutedViolationData).nodeRefutedTime,
+      }
+
+    case ViolationType.SyncingTooLong:
+      return {
+        isDuplicate:
+          nodeAccount.nodeAccountStats.lastPenaltyTime ===
+          (tx.violationData as SyncingTimeoutViolationData).nodeDroppedTime,
+        eventTime: (tx.violationData as SyncingTimeoutViolationData).nodeDroppedTime,
+      }
+
+    default:
+      throw new Error(`Unknown Violation type: , ${tx.violationType}`)
+  }
+}
+
```

Comment:
And instead of checking if the lastPenaltyTime equals with eventTime, maybe we should check by lastPenaltyTime >= eventTime. ( That way, we only apply the new event tx with higher timestamp. )